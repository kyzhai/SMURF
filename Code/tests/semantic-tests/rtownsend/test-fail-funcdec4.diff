1,18c1
< Program: 
< 	SFuncdec: 
< 		f 0 m :: Unknown = 6
< 	New Scope: 
< 		ID: m :: Unknown
< 
< 	
< 	SFuncdec: 
< 		f 0 a :: Unknown = 4
< 	New Scope: 
< 		ID: a :: Unknown
< 
< 	
< Global Scope: 
< 	ID: f :: Int -> Int -> Int
< 	ID: print :: Unknown
< 	ID: random :: Unknown
< 
---
> Fatal error: exception Sast.Multiple_identical_pattern_lists("0 m")
