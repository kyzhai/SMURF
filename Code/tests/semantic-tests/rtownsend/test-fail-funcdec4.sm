/* Pattern lists must be unique for multiple function declarations */
f :: Int -> Int -> Int

f 0 a = 4
f 0 m = 6

main = []
