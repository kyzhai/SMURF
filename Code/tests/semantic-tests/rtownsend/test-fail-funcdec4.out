Fatal error: exception Sast.Main_missing
